File: python_ruby/instance.py
class C(object):
    def __init__(self, v) :
        self.value = v
    def show(self) :
        print(self.value)
    def setValue(self, v) :
        self.value = v
    def getValue(self) :
        return self.value
c1 = C(10)
print(c1.value)
c1.value=20
print(c1.value)
c1.show()
c1.setValue(55)
print(c1.getValue())


class D(object) :
    def __init__(self, v):
        self.__value = v
    def show(self):
        print(self.__value)
    def setValue(self, v):
        self.__value = v
d1 = D(100)
# print(d1.__value) # can't access
d1.show()
d1.setValue(555)
d1.show()

File: python_ruby/instance.rb
class C
    def initialize(v)
        @value = v
    end
    def show()
        p @value
    end
    def setValue(v)
        @value = v
    end
    def getValue()
        return @value
    end
end

c1 = C.new(10)
# p c1.value  # In ruby can't access variable in instance
# c1.value = 20
# p c1.value
c1.show()
c1.setValue(55)
p c1.getValue()

class D
    attr_reader :value
    attr_writer :value
    attr_accessor :value2
    def initialize(v)
        @value = v
        @value2 = v+1
    end
    def show()
        p @value
    end
end

d1 = D.new(555)
p d1.value
d1.value = 777
p d1.value